pkg/machine/e2e/list_test.go

```diff
@@ -70,6 +70,34 @@ var _ = Describe("podman machine list", func() {
 		Expect(util.StringInSlice(name1, listNames)).To(BeTrue())
 		Expect(util.StringInSlice(name2, listNames)).To(BeTrue())
 	})
+
+	It("list machine: check if running while starting", func() {
+		i := new(initMachine)
+		session, err := mb.setCmd(i.withImagePath(mb.imagePath)).run()
```

Comment:
if you want to avoid the start later in the code, you can use .WithNow(), which will start the machine as part of init.

Reply:
I tried this and that (for some reason) reintroduces the race condition. Maybe because `init --now` takes a fraction of a second longer to exit than `start`? Not sure but I am going to keep it with start.